### DISCUSSION TITLE
How to change the text color of a disabled CheckBox

### DISCUSSION THREAD

**Discussion by wgraaf** [AvaloniaUI/Avalonia#17146] [Q&A] [answered]
The text of a checkbox does not look disabled, at least not in the simple theme.
How do I create a style that sets the text color when disabled?

I tried:

```
    <Style Selector="CheckBox:disabled /template/ ContentPresenter">
      <Setter Property="Foreground" Value="{DynamicResource ThemeControlMidBrush}"/>
    </Syle>
```

and some variations, but I get a compile error:
Unable to resolve suitable regular or attached property Selector on type :.{Unknown type} Line 112, position 6.	WinPanel	C:\WinPanel\WinPanel\App.axaml	112

**Comment by stevemonaco** [comment, marked as answer]
`<Syle` should be `<Style`. You might want to try simply:
```xml
<Style Selector="CheckBox:disabled">
  <Setter Property="Foreground" Value="{DynamicResource ThemeControlMidBrush}"/>
</Style>
```

**Reply by wgraaf** [reply]
Duh!

**Reply by stevemonaco** [reply]
For context, this is the process:
1. Check where the `Foreground` (or `TextElement.Foreground) is being assigned:
https://github.com/AvaloniaUI/Avalonia/blob/432fbe8a9eb2e3c807bf71bed43968efbd5a3dec/src/Avalonia.Themes.Simple/Controls/CheckBox.xaml#L43-L53
2. Notice that it's a `TemplateBinding` that forwards from the `CheckBox.Foreground`.
3. Double-check that no existing style selectors are targeting this or you may need to ensure your setter has more precedence.
4. If not, just set the `CheckBox.Foreground` directly when `:disabled`.
